soil_api/dependencies/queryparams.py

```diff
@@ -1,54 +1,152 @@
-from datetime import date
-from typing import Annotated
+from typing import Annotated, List
 
-from fastapi import Depends, HTTPException, Query
+from fastapi import Depends, Query
 
+from soil_api.models.soil_property import (
+    SoilDepthLabels,
+    SoilPropertiesCodes,
+    SoilPropertyValueTypes,
+)
 from soil_api.utils.validation_helpers import (
-    validate_coordinates,
-    validate_depth,
-    validate_property,
+    validate_bbox,
+    validate_depths,
+    validate_properties,
+    validate_values,
 )
 
-# Define a union type for the dependency
-LocationQuery = tuple[float, float]
-
 
 def location_query_dependency(
-    lon: Annotated[float, Query(title="lon", description="Longitude", example=9.58)],
-    lat: Annotated[float, Query(title="lat", description="Latitude", example=60.10)],
-) -> LocationQuery:
-    validate_coordinates(latitude=lat, longitude=lon)
+    lon: Annotated[
+        float,
+        Query(title="lon", description="Longitude", example=9.58, ge=-180, le=180),
+    ],
+    lat: Annotated[
+        float, Query(title="lat", description="Latitude", example=60.10, ge=-90, le=90)
+    ],
+) -> tuple[float, float]:
     return lat, lon
 
 
-LocationQueryDep = Annotated[LocationQuery, Depends(location_query_dependency)]
+LocationQueryDep = Annotated[tuple[float, float], Depends(location_query_dependency)]
+
+
+def soil_type_top_k_dependency(
+    top_k: Annotated[
+        int,
+        Query(
+            title="top_k",
+            description="Number of most probable soil types that will be returned, sorted by probability in descending order",
+            ge=0,
+            le=30,
+            example=0,
+        ),
+    ] = 0,
+) -> int:
+    return top_k
+
+
+SoilTypeTopKDep = Annotated[int, Depends(soil_type_top_k_dependency)]
 
 
-def depth_and_property_query_dependency(
-    depth: Annotated[
-        str,
+def depth_dependency(
+    depths: Annotated[
+        List[str],
         Query(
-            title="depth range",
-            description="Depth range in cm. One of: 0-5, 5-15, 15-30, 30-60, 60-100, or 100-200. For property 'ocs' only 0-30 is available.",
-            example="0-5",
+            title="depths to include",
+            description="List of depths to include in the query.",
+            min_length=1,
+            example=list(SoilDepthLabels.__members__.values()),
+        ),
+    ]
+) -> List[str]:
+    validate_depths(depths)
+    return depths
+
+
+DepthQueryDep = Annotated[List[str], Depends(depth_dependency)]
+
+
+def property_dependency(
+    properties: Annotated[
+        List[str],
+        Query(
+            title="properties to include",
+            description="List of soil properties to include in the query.",
+            min_length=1,
+            example=list(SoilPropertiesCodes.__members__.values()),
+        ),
+    ]
+) -> List[str]:
+    validate_properties(properties)
```

Comment:
Like with the DepthLabels, I think you can let FastAPI/Pydantic handle the validation if you specify properties as list[SoilPropertiesCodes] instead of list[str]. Then you also won't need the validation code